### PR TITLE
1.1 Updates

### PR DESCRIPTION
**Pull request by phenning**
@joeloff @pranavkm @mlorbetske 

Razor Precompilation package + other updates.

### CODE REVIEW

**Comment by phenning** [comment]
Do we also want this for WebAPI?

@DamianEdwards

File: src/Templates.xml [hunk 1]
```diff
@@ -394,17 +397,18 @@
       <Package Name="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="1.1.0" />
       <Package Name="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="1.1.0"/>
       <Package Name="Microsoft.AspNetCore.Mvc" Version="1.1.0" />
+      <Package Name="Microsoft.AspNetCore.Mvc.Razor.ViewCompilation" Version="1.1.0-msbuild3-final" />
       <Package Name="Microsoft.AspNetCore.StaticFiles" Version="1.1.0" />
       <Package Name="Microsoft.EntityFrameworkCore.Design" Version="1.1.0" />
       <Package Name="Microsoft.EntityFrameworkCore.SqlServer" Version="1.1.0" />
       <Package Name="Microsoft.EntityFrameworkCore.SqlServer.Design" Version="1.1.0" />
-      <Package Name="Microsoft.EntityFrameworkCore.Tools" Version="1.0.0-msbuild3-final" />
+      <Package Name="Microsoft.EntityFrameworkCore.Tools" Version="1.1.0-msbuild3-final" />
```
Comment (pranavkm) [inline src/Templates.xml:405]: cc @bricelam  to verify

**Review by pranavkm — APPROVED** [review]

File: src/Templates.xml (hunk 1 above)
Comment (bricelam) [reply src/Templates.xml:405]: 👍 LGTM

File: src/Templates.xml [hunk 2]
```diff
@@ -394,17 +397,18 @@
       <Package Name="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="1.1.0" />
       <Package Name="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="1.1.0"/>
       <Package Name="Microsoft.AspNetCore.Mvc" Version="1.1.0" />
+      <Package Name="Microsoft.AspNetCore.Mvc.Razor.ViewCompilation" Version="1.1.0-msbuild3-final" />
       <Package Name="Microsoft.AspNetCore.StaticFiles" Version="1.1.0" />
       <Package Name="Microsoft.EntityFrameworkCore.Design" Version="1.1.0" />
```
Comment (bricelam) [inline src/Templates.xml:402]: This could be removed. It's brought in by `Microsoft.EntityFrameworkCore.Tools` now.

**Comment by bricelam** [comment]
Nit: `Microsoft.EntityFrameworkCore.Design`, `Microsoft.EntityFrameworkCore.SqlServer.Design` & `Microsoft.EntityFrameworkCore.Tools` should have `PrivateAssets="All"` so they don't get published.

**Comment by pranavkm** [comment]
> Nit: Microsoft.EntityFrameworkCore.Design, Microsoft.EntityFrameworkCore.SqlServer.Design & Microsoft.EntityFrameworkCore.Tools should have PrivateAssets="All" so they don't get published.

Oh right. That's true for `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` too.

**Comment by bricelam** [comment]
Oh wait, this isn't an MSBuild file... 😁 Ignore that if it doesn't make sense here.

**Comment by phenning** [comment]
We should actually add PrivateAsset in this case. I'll need to update the rules engine to support it. Will update PR.

File: src/Templates.xml (hunk 2 above)
Comment (phenning) [reply src/Templates.xml:402]: @bricelam Will look at this for RTW.

**Review by bricelam — APPROVED** [review]
:shipit:

**Pull request merged**